accessibility/css/validation.js

```diff
@@ -40,6 +40,7 @@ function createLink(testItem) {
   anchor.href = '#' + testItem.input.name;
   anchor.onclick = function() {
     testItem.input.focus();
+    return false;
```

Comment:
1. According to [this SO question](https://stackoverflow.com/questions/128923/whats-the-effect-of-adding-return-false-to-a-click-event-listener) one should use `event.preventDefault()` instead of returning `false`.
2. Since this is a workaround until the [mentioned bug](https://bugzilla.mozilla.org/show_bug.cgi?id=277178) would be fixed, I would add a comment to remove it once the bug is fixed

Reply:
is the mentioned bug open for 16 years or am I imagining?

Reply:
Actually I wouldn't bother changing the code at all (it's a workaround).
You might want to edit the documentation to mention that adding `e.preventDefault()` after focusing is a workaround for FF, so students can play with it by themselves.

Reply:
> Actually I wouldn't bother changing the code at all (it's a workaround).
> You might want to edit the documentation to mention that adding `e.preventDefault()` after focusing is a workaround for FF, so students can play with it by themselves.

Actually its not a workaround, we don't want the browser to do anything after the focus command
We don't need the browser to follow the link and also to not propagate.
And returning false will work in any browser and it will be forward compatible with any browser changes

Reply:
I looked at [the specs](https://html.spec.whatwg.org/multipage/browsing-the-web.html#scroll-to-fragid) - A link should make the input focus without any explicit `focus()`.
I think safari handles it correctly - according to [this SO answer](https://stackoverflow.com/a/52902953/2719361) - but I haven't checked it, you are welcome to check it and change the docs accordingly.
We are left with FF - so yeah it _is_ a workaround for a bug (Spec is only what counts).

Regarding `return false` I didn't find a spec so I wouldn't rush to include it in the code.

Reply:
> I looked at [the specs](https://html.spec.whatwg.org/multipage/browsing-the-web.html#scroll-to-fragid) - A link should make the input focus without any explicit `focus()`.

Thanks alot for your engagement in this pull request
Could you please quote the specific sentence in the spec that says that
> A link should make the input focus without any explicit `focus()`.
?